src/main/java/ch/heigvd/pro/b04/participants/Participant.java

```diff
@@ -1,22 +1,44 @@
 package ch.heigvd.pro.b04.participants;
 
+import ch.heigvd.pro.b04.sessions.UserToken;
+import javax.persistence.Column;
 import javax.persistence.EmbeddedId;
 import javax.persistence.Entity;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 
 @Entity
 @NoArgsConstructor
+@AllArgsConstructor
+@Builder
 public class Participant {
+  /**
+   * Creates a new unique identifier for a Participant.
+   * @param repository The repository to which we will add a new Participant to
+   * @return A new unique identifier
+   */
+  public static Long getNewIdentifier(ParticipantRepository repository) {
+    Long identifier = repository.findAll().stream()
+        .map(Participant::getIdParticipant)
+        .map(ParticipantIdentifier::getIdParticipant)
+        .max(Long::compareTo)
+        .map(id -> id + 1)
+        .orElse(1L);
+    return identifier;
+  }
 
   @Getter
   @EmbeddedId
   private ParticipantIdentifier idParticipant;
 
-  private String username;
+  @Column(unique = true)
+  @Getter
+  private String token;
 
-  public Participant(long id, String username) {
+  public Participant(long id, UserToken token) {
```

Comment:
When using this constructor, don't we lose information related to the `Session` to which the participant belongs ?